### ISSUE TITLE
Incorrect validation error

### ISSUE THREAD

**Issue by bhavitvyamalik** [huggingface/huggingface_hub#1343]
### Describe the bug

I'm not able to push my adapter model to hub because of incorrect validation error:

> huggingface_hub.utils.validators.HFValidationError: Repo id must use alphanumeric chars or ‘-’, '', ‘.’, ‘–’ and ‘…’ are forbidden, ‘-’ and ‘.’ cannot start or end the name, max length is 96: ‘aXFQcyOEpksZmnbENRFdTmaKaSBhTYFjRfYIgsuTegBoJlhMjqzzsScRkstravxQFPkmstYSjKcEZZmMVFaYRBOHilKgotGBWwqTFBhzVJrVpNzOPLjyYMKXkYeqKkxj’.

my repo name is `joint_dt_fiction_travel` which is perfectly valid but it's not letting me push models because of invalid name error. This is how I'm pushing the model:

```
model.push_adapter_to_hub(
        repo_name= "joint_dt_fiction_travel",
        adapter_name= f"{adapter_name}",
        organization="UDApter",
        adapterhub_tag="nli/multinli",
        datasets_tag="multi_nli"
    )
```
Any help will be appreciated. Thanks!

### Logs

```
Some weights of the model checkpoint at bert-base-uncased were not used when initializing BertModelWithHeads: ['cls.predictions.decoder.weight', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'cls.seq_relationship.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.bias', 'cls.seq_relationship.bias']
- This IS expected if you are initializing BertModelWithHeads from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertModelWithHeads from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Traceback (most recent call last):
  File "push_to_hub.py", line 62, in <module>
    model.push_adapter_to_hub(
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/transformers/adapters/hub_mixin.py", line 155, in push_adapter_to_hub
    repo_url = self._get_repo_url_from_name(
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/transformers/file_utils.py", line 2056, in _get_repo_url_from_name
    return HfApi(endpoint=HUGGINGFACE_CO_RESOLVE_ENDPOINT).create_repo(
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
    validate_repo_id(arg_value)
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/huggingface_hub/utils/_validators.py", line 172, in validate_repo_id
    raise HFValidationError(
huggingface_hub.utils._validators.HFValidationError: Repo id must use alphanumeric chars or '-', '_', '.', '--' and '..' are forbidden, '-' and '.' cannot start or end the name, max length is 96: 'aXFQcyOEpksZmnbENRFdTmaKaSBhTYFjRfYIgsuTegBoJlhMjqzzsScRkstravxQFPkmstYSjKcEZZmMVFaYRBOHilKgotGBWwqTFBhzVJrVpNzOPLjyYMKXkYeqKkxj'.
```

### System info

```shell
- huggingface_hub version: 0.12.0
- Platform: Linux-5.8.0-59-generic-x86_64-with-glibc2.29
- Python version: 3.8.10
- Running in iPython ?: No
- Running in notebook ?: No
- Running in Google Colab ?: No
- Token path ?: /home/bhavitvya/.cache/huggingface/token
- Has saved token ?: True
- Who am I ?: bhavitvyamalik
- Configured git credential helpers: 
- FastAI: N/A
- Tensorflow: N/A
- Torch: 1.9.1+cu111
- Jinja2: N/A
- Graphviz: N/A
- Pydot: N/A
- Pillow: N/A
- hf_transfer: N/A
- ENDPOINT: https://huggingface.co
- HUGGINGFACE_HUB_CACHE: /home/bhavitvya/.cache/huggingface/hub
- HUGGINGFACE_ASSETS_CACHE: /home/bhavitvya/.cache/huggingface/assets
- HF_HUB_OFFLINE: False
- HF_TOKEN_PATH: /home/bhavitvya/.cache/huggingface/token
- HF_HUB_DISABLE_PROGRESS_BARS: None
- HF_HUB_DISABLE_SYMLINKS_WARNING: False
- HF_HUB_DISABLE_IMPLICIT_TOKEN: False
- HF_HUB_ENABLE_HF_TRANSFER: False
```

**Comment by Wauplin** [comment]
Hi @bhavitvyamalik , I investigated a bit your issue but couldn't find the root cause for sure. For some reason, the `repo_id` is set to `"aXFQcyOEpksZmnbENRFdTmaKaSBhTYFjRfYIgsuTegBoJlhMjqzzsScRkstravxQFPkmstYSjKcEZZmMVFaYRBOHilKgotGBWwqTFBhzVJrVpNzOPLjyYMKXkYeqKkxj"` in the process which is not a valid name. Do you know from where this could come from? Maybe the beginning of a file? In any case, it seems more related to `transformers`/`adapters-transformers`.

To help debugging this, could you:
- provide which versions of `transformers`/`adapters-transformers` you have
- provide the value of `adapter_name` in your example ?
- provide a reproducible script if possible? i.e. load a minimal model and try to push it and getting the same error

Thank you in advance :)

**Comment by Wauplin** [comment]
@bhavitvyamalik Is is possible for you to upgrade `adapter-transformers` in your environment? [Latest version is 3.1.0 and version 2.1.2 seems to be 18-months old already.](https://pypi.org/project/adapter-transformers/#history). I don't know if the problem comes from here but if it can fix it, it would be nice. Same for transformers, [latest version is 4.26.1](https://pypi.org/project/transformers/#history). Not sure which one you have.

**Comment by bhavitvyamalik** [comment]
I can do that however, I remember pushing adapter last week from my own username to HF models. It worked fine!

**Comment by Wauplin** [comment]
It could be that last week you had an earlier version of `huggingface_hub` installed where some args/kwargs in `create_repo` were not the same. They have changed ~8 months ago to be less verbose. A deprecation cycle has been made and now the old version is not supported anymore. So it would not surprise me if the latest version of `huggingface_hub` doesn't work with an old version of `adapter-transformers` anymore.

In general, `huggingface_hub` is getting more and more mature and deprecations should happen less and less in the future (which means less breaking changes). In any case, we always make sure to set warnings months in advance so that dependent libraries (like `adapter-transformers`) have the time to adapt - and reduce as much as possible the friction for end users. 

TL;DR: hope that upgrading the latest version solves your issue :smile:

**Comment by bhavitvyamalik** [comment]
I re-installed everything and the error changed :D. For pushing model to organisation, I should do `transformers-cli login` and login using my organisation's username and API token right?

**Comment by Wauplin** [comment]
Yes exactly ! Run huggingface-cli login in your terminal and add a token with write permission and that's it :)

**Comment by bhavitvyamalik** [comment]
I ran `huggingface-cli login` and typed in my username (bhavitvyamalik) and User Access Token generated for write. It keeps on saying `{"error":"Invalid username or password."}`

**Comment by Wauplin** [comment]
@bhavitvyamalik Now that you have installed version `0.12.1` could you run `huggingface-cli login` again please? You only need to paste the token, not the username (0.015 was quite old version, maybe login was different back then).

And a token must look like `"hf_xxxxxxx..."`

**Comment by bhavitvyamalik** [comment]
I did but then I'm getting this error:
```
Traceback (most recent call last):
  File "push_adapter.py", line 26, in <module>
    model.push_adapter_to_hub(
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/transformers/adapters/hub_mixin.py", line 155, in push_adapter_to_hub
    repo_url = self._get_repo_url_from_name(
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/transformers/file_utils.py", line 2056, in _get_repo_url_from_name
    return HfApi(endpoint=HUGGINGFACE_CO_RESOLVE_ENDPOINT).create_repo(
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/huggingface_hub/utils/_validators.py", line 124, in _inner_fn
    return fn(*args, **kwargs)
TypeError: create_repo() got an unexpected keyword argument 'organization'
```

**Comment by Wauplin** [comment]
Oh ok sorry. Then it's an issue to solve in adapter-transformers since it doesn't support `huggingface_hub==0.12.1` apparently. You should be able to get it running when using `huggingface_hub==0.10` or `0.11`

Would you mind opening an issue in https://github.com/adapter-hub/adapter-transformers and tag me on it please?

**Comment by bhavitvyamalik** [comment]
With `huggingface_hub==0.10` and `0.11` I'm getting:
`huggingface_hub.utils._validators.HFValidationError: Repo id must use alphanumeric chars or '-', '_', '.', '--' and '..' are forbidden, '-' and '.' cannot start or end the name, max length is 96: 'aXFQcyOEpksZmnbENRFdTmaKaSBhTYFjRfYIgsuTegBoJlhMjqzzsScRkstravxQFPkmstYSjKcEZZmMVFaYRBOHilKgotGBWwqTFBhzVJrVpNzOPLjyYMKXkYeqKkxj'.` again :/

**Comment by Wauplin** [comment]
Hi @bhavitvyamalik , thanks for opening the issue. I'm sorry about the messy support on the issue. I think the problem has been fixed on their side even though it is not released yet. You should still be able to use it by installing from their `master` branch.

And sorry about making you test on `0.10` and `0.11`. I don't know which version is expected to work but for sure there is one (maybe 0.8 and 0.9) since it worked at some point. Hope that you will get your setup running soon :confused:

**Comment by bhavitvyamalik** [comment]
Hey @Wauplin, I recently tried it once again with updated installation for both hub and adapter-transformer and I'm still getting permission issues error:
```
Traceback (most recent call last):
  File "push_adapter.py", line 26, in <module>
    model.push_adapter_to_hub(
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/transformers/adapters/hub_mixin.py", line 187, in push_adapter_to_hub
    repo_id, token = self._create_repo(repo_id, private=private, use_auth_token=use_auth_token)
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/transformers/utils/hub.py", line 661, in _create_repo
    url = create_repo(repo_id=repo_id, token=token, private=private, exist_ok=True)
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/huggingface_hub/utils/_validators.py", line 124, in _inner_fn
    return fn(*args, **kwargs)
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/huggingface_hub/hf_api.py", line 2010, in create_repo
    hf_raise_for_status(r)
  File "/home/bhavitvya/.cache/pypoetry/virtualenvs/domadapter-VESkl83O-py3.8/lib/python3.8/site-packages/huggingface_hub/utils/_errors.py", line 318, in hf_raise_for_status
    raise HfHubHTTPError(str(e), response=response) from e
huggingface_hub.utils._errors.HfHubHTTPError: 403 Client Error: Forbidden for url: https://huggingface.co/api/repos/create (Request ID: Root=1-63f81d7e-74407db531d0014b2b8a10ef)

You don't have the rights to create a model under this namespace
```
I'm the admin of this org and I did my cli login for huggingface hub using write access token generated (I marked `Add token as git credential? (Y/n)` to n)

**Comment by Wauplin** [comment]
Hmm ok. That's weird but at least we are progressing! Previous errors were more related to incompatibilities. Here it really seems related to permissions.

Could you try to upload under your own user namespace? If that works, it would already be a great think.

**Comment by bhavitvyamalik** [comment]
I can push model to my repo but not to my organisation

**Comment by Wauplin** [comment]
> I can push model to my repo

🎉 Good news to have at least that solved. 

> But not to my organization

Still something to fix then. Could you tell me - what is your user and organization name please ? And what code do you have when pushing to organization (are you passing org name in `repo_id` or `organization` field?). 

And do you know that once you've pushed a model to the hub, you can change its namespace (if your case, transfer it to your organization). This can be done in the settings page of the repo or using `move_repo`. Could you try that and tell me if you are able to do it?
Hope we will figure out together what's your issue now :)

**Comment by bhavitvyamalik** [comment]
Okay, I think I got the mistake. TIL organisation name and username are different :p
Thanks a lot for your help, @Wauplin!

**Comment by Wauplin** [comment]
Great! Happy to know all your problems are solved 🎉 We can close this issue now :)
